api/api.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/wavefronthq/go-proxy/config"
+)
+
+var (
+	client     = &http.Client{Timeout: time.Second * 10}
+	pointError = errors.New("Invalid points")
+)
+
+// API interface for the agent.
+type WavefrontAPI interface {
+	GetConfig(currentMillis, bytesLeft, bytesPerMinute, currentQueueSize int64) (*config.AgentConfig, error)
+	Checkin(currentMillis int64, localAgent, pushAgent, ephemeral bool, agentMetrics []byte) (*config.AgentConfig, error)
+	PostData(workUnitId, format, pointLines string) (*http.Response, error)
+	AgentError(details string)
+	AgentConfigProcessed() error
+}
+
+type WavefrontAPIService struct {
+	ServerURL string
+	AgentID   string
+	Hostname  string
+	Token     string
+	Version   string
+}
+
+func (service *WavefrontAPIService) GetConfig(currentMillis, bytesLeft, bytesPerMinute, currentQueueSize int64) (*config.AgentConfig, error) {
+	log.Println("GetConfig")
+
+	apiURL := service.ServerURL + getConfigSuffix
```

Comment:
found this on stackoverflow:

u, err := url.Parse("http://foo")
u.Path = path.Join(u.Path, "bar.html")
s := u.String() // prints http://foo/bar.html

Reply:
Very useful! Will include this in the near future as part of a broader change to cache some of the URLs that are static for every call (GetConfig, Checkin and AgentConfigProcessed).